api/tests/create_toggle_test.go

```diff
@@ -0,0 +1,73 @@
+package tests
+
+import (
+	"context"
+	"github.com/RevittConsulting/sft/sft"
+	"github.com/stretchr/testify/assert"
+	"strings"
+	"testing"
+)
+
+func TestCreateToggle(t *testing.T) {
+	// tests:
+	// normal creation
+	// creation of pre-existing toggle
+	// creation with incorrect fields?
+
+	duplicateToggle := sft.ToggleDto{
+		FeatureName: "duplicate feature",
+		ToggleMeta: sft.ToggleMeta{
+			"key 1": "value 1",
+			"key 2": "value 2",
+		},
+		Enabled: true,
+	}
+
+	// create a duplicate entry to test against.
+	initialToggle, err := sftService.CreateToggle(context.Background(), duplicateToggle)
+	if err != nil {
+		t.Errorf("error creating initial toggle: %s", err)
+	}
+	_ = initialToggle
+
+	var tests = []struct {
```

Comment:
I would tend to call this 'scenarios' and make it map[string]struct{} - that way the string key for the map becomes the 'test name' and goland and the test runner will react to it as such (useful tip!)

Reply:
That is a nice tip! I've changed the tests accordingly.